src/ProtoRoute.hs

```diff
@@ -10,13 +11,18 @@ main = do
     let f = \_ s -> putStrLn s
     curDir <- getCurrentDirectory
     (g, _) <- startGhci "ghci" (Just curDir) f
-    let execStatement s = execStream g s f
-    execStatement "import Data.Text"
-    execStatement "import Data.ProtoLens.Encoding"
-    execStatement ":load src/ProtoExports"
-    execStatement "let sr = SearchRequest {_SearchRequest'query = pack \"test\"\
-                           \, _SearchRequest'_unknownFields = ([])}"
+    let runCmd s = execStream g s f
+    runCmd "import Data.Text"
+    runCmd "import Data.ProtoLens.Encoding"
+    runCmd ":load src/ProtoExports"
+    runCmd "test = \"someQuery\""
+    runCmd $ passArgs "test" []
     -- Value of generated msg, and its serialized representation
-    execStatement "sr"
-    execStatement "encodeMessage sr"
+    runCmd "sr"
+    runCmd "encodeMessage sr"
     stopGhci g
+
+passArgs :: String -> Data.ProtoLens.FieldSet -> String
+passArgs query fields = "let sr = SearchRequest {_SearchRequest'query = pack \
+                      \ " ++ query ++ ", _SearchRequest'_unknownFields = \
+                      \ ( " ++ show fields ++ ")}"
```

Comment:
Okay, that's a nice start! Now, let's generalise this solution.

In order to create protocol buffers message, you need to know the following:

1. Name of the constructor or message name (in our case: `SearchRequest`)
2. Named of the fields (in our case: `query`)
3. Values of the fields.

If you're aware of all this information then it's possible to create valid Haskell value. So I imagine you should have a function like:

```haskell
constructProtoMessage :: MessageName -> [(FieldName, FieldValue)] -> Text
```

`MessageName` and `FieldName` are `newtype`s over `Text`. `FieldValue` is an algebraic data type that takes structure of `protobuf` messages into consideration.

I imagine you will have the following module structure:

1. `ProtoRoute.Message`: this module contains `constructProtoMessage` function with all types for this function.
2. `ProtoRoute.Ghcid`: this modules contains function like `runGhci` that takes `Text` and prints it using `ghcid`.
3. `ProtoRoute.Main`: contains `main :: IO ()` function that calles `runGhci` with proper arguments.